### ISSUE TITLE
Better throughput estimation

### ISSUE THREAD

**Issue by justheuristic** [bigscience-workshop/petals#192]
Current swarm load-balancing relies on a single throughput implemented in #21 .

It works, but there are a few ways to make this work better:
- speedtest-based speed can be unreliable. Some users reported that speedtest does not worn
   - quick fix: if speedtest failed, warn and set throughout to 100mbps; add an env variable to set specifically network throughput
- when a server holds many layers, it is less affected by low network throughput
   - find a better way to account for num_blocks;
   - for example, min(compute_throughput, network_throughput * min(num_blocks, 5))

Addtional information that could help clients find a better chain of servers for training/inference:
   - declare compute and network throughput separately
   - [maybe] declare whether or not a peer is directly reachable (not using relay)

**Comment by borzunov** [comment]
I believe a correct way to account for a server holding many blocks is:
```python
throughput = min(network_throughput, compute_throughput / max(1, num_layers / 2))
```

**Comment by borzunov** [comment]
This was mostly solved in #314 and #316.
